src/h3/mod.rs

```diff
@@ -26,4 +26,778 @@
+
+    /// QPACK decoder stream error.
+    QpackDecoderStreamError = -27,
+}
+
+impl Error {
+    pub fn to_wire(self) -> u16 {
+        match self {
+            Error::Done => 0x0,
+            Error::WrongSettingDirection => 0x1,
+            Error::PushRefused => 0x2,
+            Error::InternalError => 0x3,
+            Error::PushAlreadyInCache => 0x4,
+            Error::RequestCancelled => 0x5,
+            Error::IncompleteRequest => 0x6,
+            Error::ConnectError => 0x07,
+            Error::ExcessiveLoad => 0x08,
+            Error::VersionFallback => 0x09,
+            Error::WrongStream => 0xA,
+            Error::LimitExceeded => 0xB,
+            Error::DuplicatePush => 0xC,
+            Error::UnknownStreamType => 0xD,
+            Error::WrongStreamCount => 0xE,
+            Error::ClosedCriticalStream => 0xF,
+            Error::WrongStreamDirection => 0x10,
+            Error::EarlyResponse => 0x11,
+            Error::MissingSettings => 0x12,
+            Error::UnexpectedFrame => 0x13,
+            Error::RequestRejected => 0x14,
+            Error::GeneralProtocolError => 0xFF,
+            Error::MalformedFrame => 0x10,
+
+            Error::QpackDecompressionFailed => 0x20, // TODO: value is TBD
+            Error::QpackEncoderStreamError => 0x21,  // TODO: value is TBD
+            Error::QpackDecoderStreamError => 0x22,  // TODO: value is TBD
+            Error::BufferTooShort => 0x999,
+        }
+    }
+}
+
+impl std::convert::From<super::Error> for Error {
+    fn from(err: super::Error) -> Self {
+        match err {
+            super::Error::Done => Error::Done,
+            super::Error::BufferTooShort => Error::BufferTooShort,
+            _ => Error::GeneralProtocolError,
+        }
+    }
+}
+
+/// An HTTP/3 configuration.
+pub struct Config {
+    num_placeholders: u64,
+    max_header_list_size: u64,
+    qpack_max_table_capacity: u64,
+    qpack_blocked_streams: u64,
+}
+
+impl Config {
+    pub fn new(
+        num_placeholders: u64, max_header_list_size: u64,
+        qpack_max_table_capacity: u64, qpack_blocked_streams: u64,
+    ) -> Result<Config> {
+        Ok(Config {
+            num_placeholders,
+            max_header_list_size,
+            qpack_max_table_capacity,
+            qpack_blocked_streams,
+        })
+    }
+
+    pub fn set_num_placeholders(&mut self, num_placeholders: u64) {
+        self.num_placeholders = num_placeholders;
+    }
+
+    pub fn set_max_header_list_size(&mut self, max_header_list_size: u64) {
+        self.max_header_list_size = max_header_list_size;
+    }
+
+    pub fn set_qpack_max_table_capacity(
+        &mut self, qpack_max_table_capacity: u64,
+    ) {
+        self.qpack_max_table_capacity = qpack_max_table_capacity;
+    }
+
+    pub fn set_qpacked_blocked_streams(&mut self, qpack_blocked_streams: u64) {
+        self.qpack_blocked_streams = qpack_blocked_streams;
+    }
+}
+
+type StreamMap = BTreeMap<u64, stream::Stream>;
```

Comment:
This type is pretty short and it's only used once, so you can just inline this in the struct definition and then just use `BTreeMap::new()` when initializing it.